protocol-designer/src/ui/labware/selectors.ts

```diff
@@ -73,66 +110,86 @@ export const getLabwareOptions: Selector<Options> = createSelector(
         )
 
         const isAdapter = labwareEntity.def.allowedRoles?.includes('adapter')
-        const isOffDeck = getLabwareOffDeck(
+        const nickName = getNickNames(
+          nicknamesById,
           initialDeckSetup,
-          savedStepForms ?? {},
-          labwareId
+          labwareId,
+          savedStepForms
         )
 
-        const moduleOnDeck = getModuleUnderLabware(
-          initialDeckSetup,
-          savedStepForms ?? {},
-          labwareId
+        //  filter out moving trash, adapters, and labware in
+        //  waste chute for moveLabware
+        return isAdapter || isLabwareInWasteChute
+          ? acc
+          : [
+              ...acc,
+              {
+                name: nickName,
+                value: labwareId,
+              },
+            ]
+      },
+      []
+    )
+    return _sortLabwareDropdownOptions(moveLabwareOptions)
+  }
+)
+
+/** Returns options for labware dropdowns for moveLiquids.
+ * Ordered by display name / nickname, but with trash at the bottom.
+ */
+export const getLabwareOptions: Selector<Options> = createSelector(
+  stepFormSelectors.getLabwareEntities,
+  getLabwareNicknamesById,
+  stepFormSelectors.getInitialDeckSetup,
+  stepFormSelectors.getSavedStepForms,
+  stepFormSelectors.getAdditionalEquipmentEntities,
+  (
+    labwareEntities,
+    nicknamesById,
+    initialDeckSetup,
+    savedStepForms,
+    additionalEquipmentEntities
+  ) => {
+    const wasteChuteLocation = Object.values(additionalEquipmentEntities).find(
+      aE => aE.name === 'wasteChute'
+    )?.location
+    const labwareOptions = reduce(
+      labwareEntities,
+      (
+        acc: Options,
+        labwareEntity: LabwareEntity,
+        labwareId: string
+      ): Options => {
+        const isLabwareInWasteChute = Object.values(savedStepForms).find(
+          form =>
+            form.stepType === 'moveLabware' &&
+            form.labware === labwareId &&
+            form.newLocation === wasteChuteLocation
         )
-        const module =
-          moduleOnDeck != null ? getModuleShortNames(moduleOnDeck.type) : null
 
-        const isLabwareInColumn4 = getLabwareInColumn4(
+        const isAdapter = labwareEntity.def.allowedRoles?.includes('adapter')
```

Comment:
isAdapter maybe undefined

```suggestion
        const isAdapter = labwareEntity.def.allowedRoles?.includes('adapter') || false
```